### ISSUE TITLE
[Bug]: Remove tabindex from header and floating filter when are empty for accessibility

### ISSUE THREAD

**Issue by pedroslp** [ceolter/ag-grid#12883]
### Link to reproducible scenario

This is custom table, but can check it in the images

### Describe the bug

Steps to reproduce 
1. Each column is a custom button component
2. Set headerName as empty = '' for ColDef
3. Also Sortable: false, filter: false, floatingFilter: false

Actual behaviour
As each button is a new column, each one shows the outline when using the tab when the column and filter are empty. This behavior violates WCAG 2.4.3 (Focus Order)

Expected behaviour
Only interactive or meaningful elements should be focusable and included in the tab order

Maybe make it configurable, add a new property to ColDef to suppress the tab for header (supressTabIndexHeader) and filters (supressTabIndexFilter)

More information

<img width="378" height="380" alt="Image" src="https://github.com/user-attachments/assets/4205b8c3-be45-421a-b223-c713b20b9e15" />

<img width="368" height="326" alt="Image" src="https://github.com/user-attachments/assets/8c50e23c-fabe-4855-8ce3-cfcbfb183104" />

### Version

ag-grid-angular - v34.3.0

### Does the issue occur for a specific framework only?

All languages

### Is the issue only observable on a specific browser?

All browsers
